testing.go

```diff
@@ -61,19 +60,61 @@ func AwaitTimeoutInsert[Type any, Status StatusType](t testing.TB, w *Workflow[T
 	}
 }
 
-func Require[Type any, Status StatusType](t testing.TB, w *Workflow[Type, Status], foreignID string, waitFor Status, expected Type) {
+func Require[Type any, Status StatusType](t testing.TB, w *Workflow[Type, Status], foreignID string, waitForStatus Status, expected Type) {
 	if t == nil {
 		panic("Require can only be used for testing")
 	}
 
-	if !w.statusGraph.IsValid(int(waitFor)) {
-		t.Error(fmt.Sprintf(`Status provided is not configured for workflow: "%v" (Workflow: %v)`, waitFor, w.Name))
+	if !w.statusGraph.IsValid(int(waitForStatus)) {
+		t.Error(fmt.Sprintf(`Status provided is not configured for workflow: "%v" (Workflow: %v)`, waitForStatus, w.Name))
 		return
 	}
 
+	wr := waitFor(t, w, foreignID, func(r *Record) (bool, error) {
+		return r.Status == int(waitForStatus), nil
+	})
+
+	var actual Type
+	err := Unmarshal(wr.Object, &actual)
+	require.Nil(t, err)
+
+	// Due to nuances in encoding libraries such as json with the ability to implement custom
+	// encodings the marshaling and unmarshalling of an object could result in a different output
+	// than the one provided unbeknown to the user. Calling Marshal and Unmarshal on `expected`
+	// means that the same operations take place on the type and thus the unmarshaled versions
+	// should match.
+	encoded, err := Marshal(&expected)
+	require.Nil(t, err)
+
+	var normalisedExpected Type
+	err = Unmarshal(encoded, &normalisedExpected)
+	require.Nil(t, err)
+
+	require.Equal(t, normalisedExpected, actual)
```

Comment:
Are the fields that change unexported? Was wondering whether this would work instead of marshalling and unmarshalling:
```go
require.Empty(t, cmp.Diff(expected, actual, cmpopts.IgnoreUnexported(actual{})))
```

Reply:
The marshaling and unmarshaling is due to the method `Marshal` that a type could have and the potential for this to have an impact on any field. This is the best way to replicate those situations.